util/labels/helpers.go

```diff
@@ -37,3 +37,14 @@ func HasWatchLabel(o metav1.Object, labelValue string) bool {
 	}
 	return val == labelValue
 }
+
+// SetObjectLabel sets the given key to "true" on the object's labels.
+func SetObjectLabel(o metav1.Object, key string) {
```

Comment:
Gotcha, WDYT about making a generic SetObjectLabel function that adds any key/value instead of setting it to true? The main motivation for that was to avoid having check to initialize a nil map every time we want to add label.